Options.py

```diff
@@ -900,6 +901,217 @@ class ItemSet(OptionSet):
+    def from_any(cls, data: PlandoTextsFromAnyType) -> Self:
+        texts: typing.List[PlandoText] = []
+        if isinstance(data, typing.Iterable):
+            for text in data:
+                if isinstance(text, typing.Mapping):
+                    if random.random() < float(text.get("percentage", 100)/100):
+                        at = text.get("at", None)
+                        if at is not None:
+                            texts.append(PlandoText(
+                                at,
+                                text.get("text", ""),
+                                text.get("percentage", 100)
+                            ))
+                elif isinstance(text, PlandoText):
+                    if random.random() < float(text.percentage/100):
+                        texts.append(text)
+                else:
+                    raise Exception(f"Cannot create plando text from non-dictionary type, got {type(text)}")
+            cls.verify_keys([text.at for text in texts])
+            return cls(texts)
+        else:
+            raise NotImplementedError(f"Cannot Convert from non-list, got {type(data)}")
+
+    @classmethod
+    def get_option_name(cls, value: typing.List[PlandoText]) -> str:
+        return str({text.at: text.text for text in value})
+
+    def __iter__(self) -> typing.Iterator[PlandoText]:
+        yield from self.value
+
+    def __getitem__(self, index: typing.SupportsIndex) -> PlandoText:
+        return self.value.__getitem__(index)
+
+    def __len__(self) -> int:
+        return self.value.__len__()
+
+
+class ConnectionsMeta(AssembleOptions):
+    def __new__(mcs, name, bases, attrs):
+        if name != "PlandoConnections":
+            assert "entrances" in attrs, f"Please define valid entrances for {name}"
+            attrs["entrances"] = frozenset((connection.lower() for connection in attrs["entrances"]))
+            assert "exits" in attrs, f"Please define valid exits for {name}"
+            attrs["exits"] = frozenset((connection.lower() for connection in attrs["exits"]))
+        if "__doc__" not in attrs:
+            attrs["__doc__"] = PlandoConnections.__doc__
+        cls = super().__new__(mcs, name, bases, attrs)
+        return cls
+
+
+class PlandoConnection(typing.NamedTuple):
+    entrance: str
+    exit: str
+    direction: str  # entrance, exit or both
+    percentage: int = 100
+
+
+PlandoConFromAnyType = typing.Union[
+    typing.Iterable[typing.Union[typing.Mapping[str, typing.Any], PlandoConnection, typing.Any]], typing.Any
+]
+
+
+class PlandoConnections(Option[typing.List[PlandoConnection]], metaclass=ConnectionsMeta):
+    """Generic connections plando. Format is:
+    - entrance: "Entrance Name"
+      exit: "Exit Name"
+      direction: "Direction"
+      percentage: 100
+    Direction must be one of 'entrance', 'exit', or 'both', and defaults to 'both' if omitted.
+    Percentage is an integer from 1 to 100, and defaults to 100 when omitted."""
+    class Direction:
```

Comment:
if this is going to be a scoped class like this it should probably be in the PlandoConnection one instead of here so that the direction can be correctly typed?

Reply:
I agree with this.

But also, maybe an `Enum`?

(The downside to making it an `Enum` is the long definition... `PlandoConnection.Direction.both` compared to `"both"`)

So another alternative is no class, just a type alias:
```python
PlandoConnectionDirection = typing.Literal["entrance", "exit", "both"]


class PlandoConnection(typing.NamedTuple):
    entrance: str
    exit: str
    direction: PlandoConnectionDirection
    percentage: int = 100
```
(I think a type alias has to be gloabal scope.)

Reply:
Once 3.8 is dropped, StrEnum would probably be a good option. Not sure what the best option is for right now.

Reply:
I don't see what value `StrEnum` would have over `Enum` for this.
The values in a normal `Enum` can be `str`.
`StrEnum` is for when you want to pass it to a function that only takes `str`, or `str` methods like `lower`, but I don't see you doing any of that here.

Reply:
It almost seems better to make the values:
```python
class Direction(Enum):
    entrance = "=>"
    exit = "<="
    both = "<=>"
```
so that you could use those values in `get_option_name` (and simplify the code there)

Reply:
the issue with that is that the existing implementations look specifically for specifically the string when resolving the connections, so it'd need to be an alias of some sort.

Reply:
Ok, I didn't realize this was connecting with past implementations.